Feature refactor helm upgrade

aiciobanu opened the pull request:
Closes #

Users will be able to upgrade existing helm deployments in a more dynamic way.

#### Changelog

**New**

- Added `helmUpgrade` task, which will upgrade helm releases based on provided parameters.

**Changed**

- Upgraded `yq` from v3 to v4
- Updated yq syntax to match the changes made in v4

**Removed**

- N/A

#### Testing / Reviewing

{{ Add descriptions, steps or a checklist for how reviewers can verify this PR works or not }}

tlawrie commented:
Hey @aiciobanu thanks for the PR and discussion. LGTM 👍 

Can you rebase onto main and make sure it all still works, in particular make the same `yq` changes on the Dockerfile-test?

aiciobanu commented:
@tlawrie Merged main into this branch and updated yq's version for Dockerfile-test.
Also added the custom Git directory parameter in case the values YAML file is not in the default directory.
